### ISSUE TITLE
Feat Request: Support ARM 

### ISSUE THREAD

**Issue by neo-neo1** [taylorbourne/xteve_lazystream#82]
This container not supporting ARM is extremely limiting. All SBCs (Rasp Pi, ODROID, etc..) are automatically canceled out. Please make this a multi-arch and support ARM . Please include support ARMV7 (32 bit) and ARMV7 (64 bit) for universal compatibility. 

Thanks in advance.

**Comment by taylorbourne** [comment]
I might be extra tired today, but I don't really think the tone of this feature request matches the environment I would like to see in an open source repository. 

I've yet to have anyone open an issue requesting that we add this support so I'm not sure how limiting we're being here. The spirit of open source lies in your ability to fork the repo and make changes that suit your personal needs. From there, feel free to open a pull request and we'll get it added here.

If you look through the repos history, this is how many bug fixes, features, etc. have made they're way in. 

That being said – if you need help getting this feature in, simply opening an issue outlining your problem would suffice. Everyone who contributes to this container and the underlying projects do so in their free time, and that time should be respected. 

> hello! Would it be possible to add arm support to future releases? Thanks!

**Comment by netty0** [comment]
> I might be extra tired today, but I don't really think the tone of this feature request matches the environment I would like to see in an open source repository.
> 

It's difficult to sense a tone in writing so I purposely used "please" 2 times and ended with a "thanks. You seem to be reading too much into it.

**Comment by neo-neo1** [comment]
> I might be extra tired today, but I don't really think the tone of this feature request matches the environment I would like to see in an open source repository.
> 

It's difficult to sense a tone in writing so I purposely used "please" 2 times and ended with a "thanks. You seem to be reading too much into it.

**Comment by taylorbourne** [comment]
I did see the pleases and thank yous, but in my interpretation they lost some of their shine after that first sentence.

Fair enough though, happy to admit that I was wrong! I will say though I haven't had this request yet. I'll have to look into what the best way is to continue supporting multiple architectures. We did add arch64 support here https://github.com/taylorbourne/xteve_lazystream/pull/49 though, you would have to build the container yourself.

**Comment by chapperino** [comment]
I've also been watching this repo seeing if armhf support would be added to throw on my rasp p :-) 

-C.A.

**Comment by tarkah** [comment]
Lazystream has `armv7-gnueabihf` and `aarch64` releases already. As @taylorbourne pointed out, we've already added `aarch64` support in #49. However, it doesn't appear that xteve has `armv7` binaries available, only `arm64`. So this can't be done until there are `armv7` binaries published for it.

**Comment by neo-neo1** [comment]
> Lazystream has `armv7-gnueabihf` and `aarch64` releases already. As @taylorbourne pointed out, we've already added `aarch64` support in #49. However, it doesn't appear that xteve has `armv7` binaries available, only `arm64`. So this can't be done until there are `armv7` binaries published for it.

Yep I tested your `armv7-gnueabihf` lazystream release and it works great. Truth be told I don't use xteve. A containerized version of lazystream is my sole desire. Thanks

**Comment by tarkah** [comment]
If you're just running lazystream, why do you need it containerized? The binary runs on its own.

**Comment by chapperino** [comment]
unfortunately raspberry pi os (formerly raspbian) is a 32bit os. 64bit is in development and not stable yet. xteve is nice but id gladly settle for a container without it

**Comment by neo-neo1** [comment]
> If you're just running lazystream, why do you need it containerized? The binary runs on its own.

Because i run a Kubernetes node. Even if i wasn't i wouldn't run it bare. Most "self hosted" apps today are distributed container first as containerization brings portability, agility, and security among many other things.

**Comment by tarkah** [comment]
I agree, but lazystream isn't a "hosted" app, so I have no plans to Dockerfile it. You should be able to trivially throw together a few line Dockerfile that downloads the latest release and runs it as the entrypoint.

**Comment by taylorbourne** [comment]
Sounds like it'd be easiest for you guys to roll your own container, this is a pretty specific bundle.

**Comment by neo-neo1** [comment]
> I have no plans to Dockerfile it. 
>

Not asking to create a new project. It's a feature request for an already existing container project.

**Comment by chapperino** [comment]
> Sounds like it'd be easiest for you guys to roll your own container, this is a pretty specific bundle.
> 

the standard template reply to why i usually stray away from creating feat req posts.

**Comment by netty0** [comment]
This useful feature request was doomed from beginning when owner took it as personally offensive and other contributor inquired why someone would possibly need a container. Majority of the world is on arm32 - but im not here to convince you.

**Comment by taylorbourne** [comment]
It was never doomed, I said my piece and it's over now. We're all okay. 

The purpose of this container is to bring live sports/TV with an XMLTV guide to the media player of your choice – it was never my purpose to simply containerize LazyStream. 

I said in my original post that if help was needed introducing ARM support I'd be happy to lend a hand (or do it myself, which is really what I meant). I went on to say I'd like to look at better ways of making that happen, though. 

I have to reiterate guys, this is the first I'm hearing of a need to support this.. we've yet to have ONE issue opened asking for it. Yet here we are with 3 requests! Give me tonight (in between caring for my newborn daughter) to look at this and see what can be done. 

Also up there I mentioned that I'm working on a rewrite of xTeVe, and I committed to getting ARM support there as well. 

Will accept any ideas you guys have about how to move forward.

**Comment by taylorbourne** [comment]
> This useful feature request was doomed from beginning when owner took it as personally offensive and other contributor inquired why someone would possibly need a container. Majority of ARM is on ARM32 - but im not here to convince you. 

Had to reply to you too.. why didn't you open a feature request if it's so important to you? How was I supposed to know how useful this would be?

**Comment by tarkah** [comment]
> This useful feature request was doomed from beginning when owner took it as personally offensive and other contributor inquired why someone would possibly need a container. 

I'm just pointing out that lazystream in a container is a bit unnecessary. It is already released for armv7 and runs standalone without any configuration. It's a simple client program that queries some APIs, downloads / parses m3u8 files and passes some args to streamlink. However, I realize this is just my personal viewpoint. I'm more than happy for others that find it useful to open a PR, but I'm not interested in adding this.

The original request of adding armv7 support to this container is still alive. @taylorbourne has noted his work on an xTeVe fork and his willingness to look into building it for armv7, which is a blocker that I originally pointed out. Until then, there isn't much to be done.
